winter/schema/generation.py

```diff
@@ -87,17 +85,22 @@ def build_response_schema(method: ComponentMethod):
     if output_serializer is not None:
         return output_serializer.class_(**output_serializer.kwargs)
 
-    type_hints = typing.get_type_hints(method.func)
-    return_value_type = type_hints.get('return', None)
+    return_value_type = method.return_value_type
+
+    if (
+        return_value_type in (None, type(None))
+        or (inspect.isclass(return_value_type) and issubclass(return_value_type, (HttpResponseBase, ResponseEntity)))
+    ):
+        return openapi.Response(description='')
+
     try:
         type_info = inspect_type(return_value_type)
-    except InspectorNotFound:
-        return openapi.Response(description='')
-    else:
-        return type_info.get_openapi_schema()
+    except InspectorNotFound as e:
+        raise InvalidReturnTypeException(method, return_value_type, str(e))
```

Comment:
I think `invalid` is not the correct word to describe the situation. Something like `CantInspectReturnType` would be more appropriate.